### DISCUSSION TITLE
max-time! only returns first result

### DISCUSSION THREAD

**Discussion by AdrickTench** [trueagi-io/metta-wam#142] [General]
`max-time!` only returns the first result but should return all (assuming it completes with the time limit). For example 
```
metta+>!(max-time! 1 (superpose (a b c)))
[a]
```
should return `[a, b, c]`

**Comment by AdrickTench** [comment]
Interesting to consider how we should handle getting some results before timing out. Do we throw time_limit_exceeded and discard results? Or can we somehow throw without losing the incremental results?

**Comment by TeamSPoon** [comment]
secretly my intent was to give results up until the time..  sicne we are using a lazy iterator... they already have the answers (we canty take them back if we wanted to) 

then at timeout should they get an exception, they can catch,?  and if they want they can continue to get more?

**Comment by TeamSPoon** [comment]
```
metta+>!(max-time! 1 (superpose (a b c)))
[a

, b

, c]
```

**Comment by TeamSPoon** [comment]
Was a bug but now that it is fixed we still have the open question about if a query choul;d be resumable after timeout
